Bottom: 66558224885382303fdb2625ea3c8ce1e821bbab
Top:    fad57335c8ae54a82332f718120ee9917b45f78f
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-02-21 23:54:18 +0000

Refresh of play-with-perspective

---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index f6da3fa..1cd165f 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -1910,11 +1910,17 @@ ghid_drawing_area_expose_cb (GtkWidget *widget,
   /* Stash the model view matrix so we can work out the screen coordinate -> board coordinate mapping */
   glGetFloatv (GL_MODELVIEW_MATRIX, (GLfloat *)last_modelview_matrix);
 
+#if 0
   /* Fix up matrix so the board Z coordinate does not affect world Z
    * this lets us view each stacked layer without parallax effects.
+   *
+   * Commented out because it breaks:
+   *   Board view "which side should I render first" calculation
+   *   Z-buffer depth occlusion when rendering component models
    */
   last_modelview_matrix[2][2] = 0.;
   glLoadMatrixf ((GLfloat *)last_modelview_matrix);
+#endif
 
   glEnable (GL_STENCIL_TEST);
   glClearColor (port->bg_color.red / 65535.,
